### PR TITLE
Stack broadcasts so that not one is lost

### LINKED ISSUE #1608: [🐛 BUG] Submit button unactive after data node lock
### What went wrong? 🤔

When using a data node viewer and scenario viewer, editing the data node will lock the scenario and finishing the edition should unblock the possibility to submit the scenario.

However, the "Submit" button is still grey and inactive even it the edition has been done. Refreshing the scenario or the page will re-activate the button.

This happens everytime editing an input data node.

### Expected Behavior

Finishing the edition should reactivate the button automatically if the scenario can be submitted without the need to refresh the scenario or the page.

### Steps to Reproduce Issue

Run this code, choose a scenario, notice that the scenario could be submitted, and edit the data to 2. The button becomes grey and is inactive.

```python

# Import necessary libraries
import pandas as pd
import taipy as tp
from taipy import Config, Scope, Frequency
from taipy.gui import notify
import time

import datetime as dt

Config.configure_job_executions(mode="standalone", max_nb_of_workers=2)

# Function to run a Dataiku scenario
def run_something(input_1, input_2):
    datetime = dt.datetime.now()
    date = dt.date(2018, 1, 1)
    int_var = 10
    string_var = "String"
    return datetime, date, int_var, string_var


data = {"toto": [i for i in range(10_000)], 
        "titi": [2*i for i in range(10_000)],
        "tata": [4*i for i in range(10_000)]}



input_1_cfg = Config.configure_data_node(
    id="input_1_data_node",
    default_data=1,
)


input_2_cfg = Config.configure_data_node(
    id="input_2_data_node",
    default_data=data,
)

datetime_cfg = Config.configure_data_node(id="datetime_data_node")
date_cfg = Config.configure_data_node(id="date_data_node")
int_cfg = Config.configure_data_node(id="int_data_node")
string_cfg = Config.configure_data_node(id="string_data_node")


# Scenario and task configuration in Taipy
scenario_task_cfg = Config.configure_task(
    id="scenario_task",
    function=run_something,
    input=[input_1_cfg, input_2_cfg],
    output=[datetime_cfg, date_cfg, int_cfg, string_cfg]
)

scenario_cfg = Config.configure_scenario(
    id="scenario",
    task_configs=[scenario_task_cfg],
    frequency=Frequency.DAILY)

data_node = None
# GUI Markdown content
scenario_md = """
<|{scenario}|scenario_selector|>

<|{data_node}|data_node|>

<|{scenario}|scenario|>
"""


def on_change(state, var_name, var_value):
    if var_name == "scenario" and isinstance(var_value, tp.Scenario):
        state.data_node = state.scenario.input_1_data_node
# Main execution block with GUI setup
if __name__ == "__main__":
    tp.Core().run()
    scenario = tp.create_scenario(scenario_cfg)

    tp.Gui(scenario_md).run(title="Bug replication", port=3248)
```

### Version of Taipy

develop and 4.0.0.dev0

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1608

```
import datetime as dt

import taipy as tp
from taipy import Config, Frequency

Config.configure_job_executions(mode="standalone", max_nb_of_workers=2)


# Function to run a Dataiku scenario
def run_something(input_1, input_2):
    datetime = dt.datetime.now()
    date = dt.date(2018, 1, 1)
    int_var = 10
    string_var = "String"
    return datetime, date, int_var, string_var


data = {
    "toto": list(range(10_000)),
    "titi": [2 * i for i in range(10_000)],
    "tata": [4 * i for i in range(10_000)],
}


input_1_cfg = Config.configure_data_node(
    id="input_1_data_node",
    default_data=1,
)


input_2_cfg = Config.configure_data_node(
    id="input_2_data_node",
    default_data=data,
)

datetime_cfg = Config.configure_data_node(id="datetime_data_node")
date_cfg = Config.configure_data_node(id="date_data_node")
int_cfg = Config.configure_data_node(id="int_data_node")
string_cfg = Config.configure_data_node(id="string_data_node")


# Scenario and task configuration in Taipy
scenario_task_cfg = Config.configure_task(
    id="scenario_task",
    function=run_something,
    input=[input_1_cfg, input_2_cfg],
    output=[datetime_cfg, date_cfg, int_cfg, string_cfg],
)

scenario_cfg = Config.configure_scenario(id="scenario", task_configs=[scenario_task_cfg], frequency=Frequency.DAILY)

data_node = None
# GUI Markdown content
scenario_md = """
<|{scenario}|scenario_selector|>

<|{data_node}|data_node|>

<|{scenario}|scenario|>
"""


def on_change(state, var_name, var_value):
    if var_name == "scenario" and isinstance(var_value, tp.Scenario):
        state.data_node = state.scenario.input_1_data_node


# Main execution block with GUI setup
if __name__ == "__main__":
    tp.Orchestrator().run()
    scenario = tp.create_scenario(scenario_cfg)

    tp.Gui(scenario_md).run(title="1608 submit inactive after data lock")

```

### CODE REVIEW

File: frontend/taipy/package.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
     "fast-deep-equal": "^3.1.3",
     "formik": "^2.2.9",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "taipy-gui": "file:../../taipy/gui/webapp"
```
Comment (FabienLelaquais) [inline frontend/taipy/package.json]: A mistake hu?

File: frontend/taipy/src/utils.ts [hunk 2]
```diff
@@ -12,8 +12,25 @@
...
 import { Theme, alpha } from "@mui/material";
 import { PopoverOrigin } from "@mui/material/Popover";
-
-import { getUpdateVar, useDynamicProperty } from "taipy-gui";
+import { ReactNode } from "react";
+
+import { BroadcastDesc, getUpdateVar, useDynamicProperty } from "taipy-gui";
+
+
+export interface CoreProps {
```
Comment (FabienLelaquais) [inline frontend/taipy/src/utils.ts:20]: 👌

File: frontend/taipy-gui/packaging/taipy-gui.d.ts [hunk 3]
```diff
@@ -323,6 +323,26 @@ export declare const createRequestUpdateAction: (
     forceRefresh?: boolean,
     stateContext?: Record<string, unknown>
 ) => Action;
+
+/**
+ * Broadcast stack definition.
+ */
+export interface BroadcastDesc {
+    /** Name of the broadcast. */
+    name: string;
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/packaging/taipy-gui.d.ts:332]: Please improve that comment a little bit... pretty please...

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Neat!

File: frontend/taipy/package.json (hunk 1 above)
Comment (FredLL-Avaiga) [reply frontend/taipy/package.json]: good catch, tx

**Review by FabienLelaquais — DISMISSED** [review]
👍

**Review by dinhlongviolin1 — APPROVED** [review]
Very good changes

**Pull request merged**